src/Turtle/Prelude.hs

```diff
@@ -1635,6 +1662,32 @@ find pattern dir = do
       file_stat <- lstat file
       return (not (PosixCompat.isSymbolicLink file_stat))
 
+-- | Filter a shell of FilePaths according to a given pattern
+findtree :: Pattern a -> Shell FilePath -> Shell FilePath
+findtree pat files = do
+  path <- files
+  Right txt <- return (Filesystem.toText path)
+  _:_ <- return (match pat txt)
+  return path
+
+{- | Check if a file was last modified after a given
+     timestamp
+-}     
+cmin :: MonadIO io => UTCTime -> FilePath -> io Bool
+cmin t file =
+  adapt <$> lstat file
+  where
+    adapt x = posixSecondsToUTCTime (modificationTime x) > t
+
+{- | Check if a file was last modified before a given
+     timestamp
+-}     
+cmax :: MonadIO io => UTCTime -> FilePath -> io Bool
+cmax t file =
+  adapt <$> lstat file
```

Comment:
It looks like Travis is failing on older versions of GHC due to `Functor` not being a super-class of `MonadIO`.  You can fix that by using `liftM` or `do` notation